pkg/scalers/metrics_api_scaler.go

```diff
@@ -149,3 +270,52 @@ func (s *metricsAPIScaler) GetMetrics(ctx context.Context, metricName string, me
 
 	return append([]external_metrics.ExternalMetricValue{}, metric), nil
 }
+
+func getMetricAPIServerRequest(meta *metricsAPIScalerMetadata) (*http.Request, error) {
+	var req *http.Request
+	var err error
+
+	if meta.enableAPIKeyAuth {
+		if header == meta.method {
+			req, err = http.NewRequest("GET", meta.url, nil)
+			if err != nil {
+				return nil, err
+			}
+
+			if len(meta.keyParamName) == 0 {
+				req.Header.Add("X-API-KEY", meta.apiKey)
+			} else {
+				req.Header.Add(meta.keyParamName, meta.apiKey)
+			}
+
+		} else if queryParam == meta.method {
+			url, _ := neturl.Parse(meta.url)
+			queryString := url.Query()
+			if len(meta.keyParamName) == 0 {
+				queryString.Set("api_key", meta.apiKey)
+			} else {
+				queryString.Set(meta.keyParamName, meta.apiKey)
+			}
+
+			url.RawQuery = queryString.Encode()
+			req, err = http.NewRequest("GET", url.String(), nil)
+			if err != nil {
+				return nil, err
+			}
+		}
```

Comment:
Is it possible that we will reach this place? With `meta.enableAPIKeyAuth` and both above `if` we may need `else` caluse?

Reply:
we'll never reach here. the default method is the header. So always one condition is true. I used else if to have better clarity of why we are doing this.

Reply:
I would do `if { } else { }` with a comment in else about fallback to default option. We have to remember that someone will look this code in future and will have to figure out it the code will reach here or no :) 

Btw. I personally I prefere `meta.method == queryParam` as this is more natural to read "(provided) meta method is queal to queryParam"

Reply:
Right! Now it will be easier for the third party user to actually read the flow with more clarity. He doesn't have to retrace the whole code sequence to understand this section.